cli/class-block-plugin-checker.php

```diff
@@ -0,0 +1,476 @@
+	 */
+	public function __construct( $slug = null ) {
+		$this->slug = $slug;
+	}
+
+	/**
+	 * Check a plugin given the URL of a Subversion or GitHub repo.
+	 * Note that only hosts listed in $allowed_hosts are permitted.
+	 *
+	 * @param string $url The URL of a Subversion or GitHub repository.
+	 * @return array A list of status items.
+	 */
+	public function run_check_plugin_repo( $url ) {
+
+		// git@github.com:sortabrilliant/jumbotron.git
+
+		if ( preg_match( '#^\w+@github[.]com:(\w+/[^.]+)\.git$#', $url, $matches ) ) {
+			$url = 'https://github.com/' . $matches[1];
+		}
+
+		// Parse the URL with whitespace and trailing / trimmed
+		$url_parts = wp_parse_url( rtrim( trim( $url ), '/' ) );
+
+		if ( empty( $url_parts ) ) {
+			$this->record_result(
+				__FUNCTION__,
+				'error',
+				sprintf( __( 'Invalid url %s' ), $url ),
+				$url
+			);
+			return $this->results;
+		}
+
+		if ( empty( $url_parts['host'] ) || !in_array( $url_parts['host'], $this->allowed_hosts ) ) {
+			$this->record_result(
+				__FUNCTION__,
+				'error',
+				sprintf( __( 'URL must be GitHub or plugins.svn.wordpress.org %s' ), $url ),
+				$url
+			);
+			return $this->results;
+		}
+
+		if ( 'plugins.svn.wordpress.org' === $url_parts['host'] ) {
+			$path_parts = explode( '/', $url_parts[ 'path' ], 3 );
+			if ( empty( $path_parts[1] ) ) {
+				$this->record_result(
+					__FUNCTION__,
+					'error',
+					sprintf( __( 'URL must be a plugin repository %s' ), $url ),
+					$url
+				);
+				return $this->results;
+			}
+
+			$this->slug = $path_parts[1];
+			$url = 'https://plugins.svn.wordpress.org/' . $path_parts[1];
+			if ( 4 === count( $path_parts )
+				&& ( 'tags' === $path_parts[2] || 'branches' === $path_parts[2] )
+				&& ( !empty( $path_parts[3] ) ) ) {
+				$url .= $path_parts[2] . '/' . $path_parts[3];
+			} else {
+				$url .= '/trunk';
+			}
+		} elseif ( 'github.com' === $url_parts['host'] ) {
+			// https://github.com/sortabrilliant/jumbotron
+			// git@github.com:sortabrilliant/jumbotron.git
+			// https://github.com/sortabrilliant/jumbotron.git
+			//
+			if ( preg_match( '#^/(\w+/[^.]+)\.git#', $url_parts[ 'path' ], $matches ) ) {
+				$url = 'https://github.com/' . $matches[1] . '.git/trunk';
+			} else {
+				$this->record_result(
+					__FUNCTION__,
+					'error',
+					sprintf( __( 'URL must be a plugin repository %s' ), $url ),
+					$url
+				);
+				return $this->results;
+			}
+		}
+
+		$url = esc_url_raw( $url );
+
+		$path = $this->export_plugin( $url );
+
+		if ( $path ) {
+			$result = $this->run_check_plugin_files( $path );
+
+			// Be very very careful!
+			shell_exec( 'rm -rf ' . escapeshellarg( $path ) );	
+
+			return $result;
+		}
+		
+	}
+
+	function export_plugin( $svn_url ) {
+
+
+		// Generate a unique tmp file directory name, but don't create it.
+		if ( $this->slug ) {
+			$path = uniqid( "/tmp/blockplugin" ) . '-' . $this->slug;
+		} else {
+			$path = uniqid( "/tmp/blockplugin" );
```

Comment:
I would replace this with `Filesystem::temp_directory( $slug_or_something )` so that the directory is cleaned up at the end of the script automatically.

Reply:
Thanks, I'd overlooked `temp_directory()`. Fixed in aa39489.